extensions/amp-story/1.0/amp-story.js

```diff
@@ -1502,40 +1497,32 @@ export class AmpStory extends AMP.BaseElement {
   }
 
   /**
-   * Update the story level stack and check for navigation adherence
+   * Updates the story navigation stack and checks for navigation adherence to
    * to the path a user takes.
    * @param {string} targetPageId
-   * @param {string} direction
-   * @return {!./amp-story-page.AmpStoryPage}
+   * @param {!NavigationDirection} direction
    * @private
    */
-  updateStoryNavigationPath_(targetPageId, direction) {
+  updateNavigationPath_(targetPageId, direction) {
+    const navigationPath = /** @type {!Array<string>} */ (this.storeService_.get(
+      StateProperty.NAVIGATION_PATH
+    ));
+
     if (direction === NavigationDirection.PREVIOUS) {
-      this.storyNavigationPath_.pop();
-      if (this.storyNavigationPath_.length > 0) {
-        const pathPrevious = this.storyNavigationPath_[
-          this.storyNavigationPath_.length - 1
-        ];
-        if (pathPrevious !== targetPageId) {
-          return this.getPageById(pathPrevious);
-        }
-      }
-    } else if (direction === NavigationDirection.NEXT) {
-      const topOfStack = this.storyNavigationPath_[
-        this.storyNavigationPath_.length - 1
-      ];
-      // If the user navigates the away from the page, the top of storyStack
-      // will be the same as ampStoryPageId in the history state.
-      if (targetPageId !== topOfStack) {
-        this.storyNavigationPath_.push(targetPageId);
-      }
+      navigationPath.pop();
     }
-    setHistoryState(
-      this.win,
-      HistoryState.NAVIGATION_PATH,
-      this.storyNavigationPath_
-    );
-    return this.getPageById(targetPageId);
+
+    // Ensures the pageId is not at the top of the stack already, which could
+    // happen when reloading a page.
```

Comment:
Nit: maybe say "can happen on initial page load (e.g. reloading a page)"

I consider "reloading" to mean actually pulling down to reload or pressing a reload option in the UI, pressing F5, etc.  But, other cases are possible, like Chrome on mobile evicted the tab from memory because you haven't used it in a while, then you came back.  Or, you reopened a link to something days later because you liked it.

Reply:
Good point, thank you